engine/camunda/src/main/java/org/camunda/bpm/delegate/InterfaceWfcHandler.java

```diff
@@ -0,0 +1,174 @@
+    }
+    matcher.appendTail(sb);
+    return sb.toString();
+}
+
+/* 
+ *  This method is used to handle the boundary message event interupting message boundary event attached to a user task
+ * @param messageBoundaryEvent 
+ * @param delegateTask
+*/
+public void handleBoundaryEvent(BoundaryEvent messageBoundaryEvent, DelegateTask delegateTask,
+        Properties properties) {
+    MessageEventDefinition messageEventDefinition = (MessageEventDefinition) messageBoundaryEvent
+            .getEventDefinitions().iterator().next();
+    Message message = messageEventDefinition.getMessage();
+    String messageName = message != null ? message.getName() : "No message";
+
+    Collection<Documentation> documentations = messageBoundaryEvent.getDocumentations();
+    String documentationText = !documentations.isEmpty() ? documentations.iterator().next().getTextContent()
+            : "No documentation";
+
+    logger.info("Handling boundary event ID from separate component: " + messageBoundaryEvent.getId()
+            + " with message: " + messageName);
+    String variableName = messageBoundaryEvent.getName();
+    String query = processQuery(documentationText, delegateTask);
+    requestData(query, delegateTask.getProcessInstanceId(), messageName, variableName, delegateTask.getId(),
+            properties);
+}
+/*
+ * This method is used to handle the data object reference in this implemetation associated with receieve task
+ * @param properties, delegateExecution
+ */
+public void dataObjectReferenceImpl(Properties properties, DelegateExecution delegateExecution) {
+    String message = "";
+    ReceiveTask receiveTask = null;
+    receiveTask = delegateExecution.getProcessEngine().getRepositoryService()
+                .getBpmnModelInstance(delegateExecution.getProcessDefinitionId())
+                .getModelElementById(delegateExecution.getCurrentActivityId());
+    message = receiveTask.getMessage().getName();
+    String variableName = "";
+    String query = "";
+        BpmnModelInstance bpmModel = delegateExecution.getProcessEngine().getRepositoryService()
+                .getBpmnModelInstance(delegateExecution.getProcessDefinitionId());
+        receiveTask = bpmModel.getModelElementById(delegateExecution.getCurrentActivityId());
+        // Get documentation from dataReference, NOTE: Assumed only 1 dataRef with a single documentation is found.
+        for (DataInputAssociation dataInputAssociation : receiveTask.getDataInputAssociations()) {
+            for (ItemAwareElement source : dataInputAssociation.getSources()) {
+                DataObjectReferenceImpl dataReference = bpmModel.getModelElementById(source.getId());
+                variableName = dataReference.getName();
+                for (Documentation documentation : dataReference.getDocumentations()) {
+                    query = documentation.getRawTextContent();
+                }
+            }
+        }
+
+    StringBuffer sb = new StringBuffer();
```

Comment:
Is this different then processQuery() ?

Reply:
It is different in the variable origin, specifically this code is for Data Object reference attached to the receive task. 
The processQuery method for task listeners. (delegateTask).

Reply:
Even though they are evaluating similar query, the origin of the variable to replaced differs in Type, the receive task for example is, DelegateExecution, and for User task is DelegateTask, that is why I simply leave the original code for the receive task as it was.